### ISSUE TITLE
Bug in interaction with MacOS spell checker. 

### ISSUE THREAD

**Issue by vassudanagunta** [typora/typora-issues#214]
The MacOS spell checker generally works within Typora. Misspelled words are correctly underlined with a dashed red line. If you go back and correct the spelling, the dashed red line will go away, EXCEPT in one case (the bug): If you misspell a word by adding an extra character. 

Repro steps.
1. Type this sentence: "Just kall me angel of the moorning, baby."
2. Note that two words are marked as misspelled: "kall" and "moorning".
3. Change the "k" on "kall" to "c". Note that the dashed red line disappears. 
4. Remove the extra "o" from "moorning". Note that the dashed red line remains even though the word is not spelled correctly.

**Comment by vassudanagunta** [comment]
@abnerlee The behavior is (or has gotten) worse than I first reported them. I just updated the Issue description.

**Comment by olets** [comment]
In addition, words once marked misspelled stay marked misspelled even after correction:

- With "Check spelling while typing" active, type "thisx is a test"
- Delete the "x"
- The red line doesn't disappear

**Comment by timlockridge** [comment]
I know this issue is rather old, but I can still reproduce it. All in-line spell checking on MacOS is essentially disabled, and it makes Typora (which is otherwise excellent) a tough sell for writing prose on the Mac.

**Comment by kostrse** [comment]
Also spell checking should ignore inline \`code\` blocks.

**Comment by abnerlee** [comment]
@kostrse That is by design, variable name is used in inline code, which is not normal English

**Comment by kostrse** [comment]
I meant it should NOT try to spell check inline code blocks.
It works correctly in Windows, but in MacOS it messes up and tries to spell check code in the inline blocks too.

**Comment by netmasteroc3** [comment]
I want to echo @timlockridge's comment from last July. I just started using Typora on my mac laptop (normally I use it on a windows desktop) and misspelled words aren't underlined at all.

Typora Version 0.9.9.24.6 (2400).

**Comment by Jarettlh** [comment]
This is not completely relevant to the original post, but since Issue #2706 was merged into this one:

I'm having the issue outlined in #2706, namely, the spellcheck feature does not work on the x64 version on Windows (version 0.9.9.83). As per the thread suggests, downgrading to 0.9.70 fixes the issue, and spellcheck works again, so I doubt this is an issue with my system. I can provide more info if needed

**Comment by abnerlee** [comment]
@Jarettlh did spellcheck be turned on (you can check the status on status bar), and in which language it is used? Also, are you using Windows 10 or 8?

**Comment by Jarettlh** [comment]
@abnerlee Yes, it's enabled and set to English (US). I've tried auto-detect language as well. Windows 10. Like I said, I don't think it's my system because my system spell check works in versions <0.9.70 and in other software

**Comment by Coderdude112** [comment]
I am having a very similar problem, with spell checker not working on any of my files,
I am using Windows 10, 64-bit version and spell check is set to `EN-US` 
Thought I would add this here in case it helps :)

**Comment by abnerlee** [comment]
@Coderdude112 Did you installed English language pack on your Windows 10, also, did “Highlight misspelled words” in Windows Settings → Devices → Typing be enabled?

**Comment by Coderdude112** [comment]
Hello, @abnerlee, Yes I have installed the English language pack and "Highlight misspelled words" is enabled

**Comment by Samar900** [comment]
> I meant it should NOT try to spell check inline code blocks.
> It works correctly in Windows, but in MacOS it messes up and tries to spell check code in the inline blocks too.

Hi,

New user here. I am using code blocks to include citations and special comments, and would very much like to have spell check enabled here. But Typora is now ignoring the code blocks completely. Can this behavior of spellchecking (or not) within the code blocks be made optional?

Thanks.

**Comment by daeh** [comment]
Still having this issue (v1.0.5) on MacOS (v11.6.2). Sometimes editing one part of a file makes the underlines elsewhere disappear.

**Comment by olets** [comment]
@kkukshtel to the degree that this works at all, it works for me with Nord. It's still unreliable though.

**Comment by kkukshtel** [comment]
@olets Just realized I mentioned the wrong theme — I'm using Quartz. Never worked there, may try Nord.

**Comment by olets** [comment]
@kkukshtel works for me with Quartz (with the bugs described above) in version 1.2.4 (6304). Don't know if it's possible to restyle system native spelling warnings, but I don't see anything in the Quartz CSS attempting to target it.

**Comment by gagomes** [comment]
Any revised update on the state of Grammarly integration? Would be great to have this integration available in typora.

**Comment by daeh** [comment]
Unreliable spellcheck decoration continues to be a persistent aggravation for me (v1.4.8, MacOS 12.6.1). I've done a complete uninstall, tried every theme... I usually have to click on a word for the text decoration to appear.

This issue was opened in 2016.

**Comment by daeh** [comment]
Seems like v1.5.5 might have made some incremental improvements, but the major issues are still present. Now, misspelled in a given paragraph get decorated when I edit that paragraph, but misspelled words in other paragraphs remain unmarked.

**Comment by grafst** [comment]
https://languagetool.org might also be interesting (it is open source)
